oecophylla/taxonomy/parser.py

```diff
@@ -0,0 +1,55 @@
+import os
+import pandas as pd
+
+
+def combine_bracken(dir_bracken_reports):
+    """Combines bracken counts for several samples into one table.
+
+    Parameters
+    ----------
+    dir_bracken_reports : str
+        Pathname of directory that contains bracken abundance estimation files
+        per samples.
+
+    Returns
+    -------
+    Pandas.DataFrame with rows for features, columns for samples.
+    Feature labels are NCBI taxonomy IDs.
+
+    Notes
+    -----
+    We don't know which files in the given directory are bracken abundance
+    estimation files.
+    Therefore, we iterate over all files and try to a) parse the file as a tab
+    separated table and b) check if the second column is named 'taxonomy_id'.
+    More information about the file format:
+    https://ccb.jhu.edu/software/bracken/index.shtml?t=manual
+    """
+    samples = []
+    # walking over the given directory.
+    # We don't know which files are actual bracken abundance estimation files.
+    for _file in next(os.walk(dir_bracken_reports))[2]:
```

Comment:
can this be replaced with `os.listdir(dir_bracken_reports)`?

Reply:
yes, but I am not sure how the input filetree will look like. Maybe there is a dir per sample and than I need to recursively go through all sub-dirs. Therefore, I think walk() is more "future safe". But we need input from @tanaes here.

Reply:
For this function, I'd recommend taking a list of bracken taxonomy reports rather than a directory. These are explicit outputs of the kraken_bracken rule, and can be accessed explicitly through the snakemake file tracking.